integration_tests/test_eip1559.py

```diff
@@ -1,103 +1,141 @@
-from .utils import ADDRS, KEYS, send_transaction, w3_wait_for_block
+from concurrent.futures import ThreadPoolExecutor, as_completed
 
+from .utils import ADDRS, DEFAULT_GAS_PRICE, KEYS, send_transaction, w3_wait_for_block
 
-def adjust_base_fee(parent_fee, gas_limit, gas_used):
+
+def adjust_base_fee(parent_fee, gas_limit, gas_used, params):
     "spec: https://eips.ethereum.org/EIPS/eip-1559#specification"
-    change_denominator = 8
-    elasticity_multiplier = 2
+    change_denominator = params["base_fee_change_denominator"]
+    elasticity_multiplier = params["elasticity_multiplier"]
     gas_target = gas_limit // elasticity_multiplier
 
     delta = parent_fee * (gas_target - gas_used) // gas_target // change_denominator
-    return parent_fee - delta
+    # https://github.com/crypto-org-chain/ethermint/blob/develop/x/feemarket/keeper/eip1559.go#L104
+    return max(parent_fee - delta, params["min_gas_price"])
+
+
+def crosscheck(cronos, geth, process):
+    providers = [cronos.w3, geth.w3]
+    with ThreadPoolExecutor(len(providers)) as exec:
+        tasks = [exec.submit(process, w3) for w3 in providers]
+        res = [future.result() for future in as_completed(tasks)]
+        assert len(res) == len(providers)
+        assert res[0] == res[1], res
+
+
+def get_params(w3, cronos):
+    if w3 != cronos.w3:
+        return {
+            "base_fee_change_denominator": 8,
+            "elasticity_multiplier": 2,
+            "min_gas_price": 0,
+        }
+    params = cronos.cosmos_cli().query_params("feemarket")["params"]
+    return {k: int(float(v)) for k, v in params.items()}
 
 
-def test_dynamic_fee_tx(cluster):
+def test_dynamic_fee_tx(cronos, geth):
```

Comment:
isn't the `cluster` will run the test case on different networks already?

Reply:
we run one by one, can be faster test concurrently

Reply:
for parallel test, we can try https://pypi.org/project/pytest-xdist/

Reply:
not sure if port will conflicts but I get error like `No module named 'hatchling'` when build after add pytest-xdist

Reply:
chain-main uses xdist, but it's indeed tricky to setup, but my point is there must be some generic way that we can do parallel testing in general, instead of refactoring the test code like this

Reply:
seems chain-main run parallel tests by group instead of `-n auto`, we might test orthogonally for geth and cronos due to the port conflicts